pkg/webhooks/validation/controller/controller.go

```diff
@@ -225,11 +225,11 @@ func newController(
 	webhookInformer := cache.NewSharedIndexInformer(
 		&cache.ListWatch{
 			ListFunc: func(opts metav1.ListOptions) (runtime.Object, error) {
-				opts.LabelSelector = fmt.Sprintf("%s=%s", label.IoIstioRev.Name, o.Revision)
+				opts.FieldSelector = fields.OneTermEqualSelector("metadata.name", c.webhookName).String()
```

Comment:
I think the old code is intentionally to handle multiple webhooks? @Monkeyanator

Reply:
Should be fine just targeting the single webhook, nothing relies on being able to handle multiple at the moment.

Reply:
I am not sure I understand, we install two now:
```
$ istioctl manifest generate | k grep 'Validat*/' -s
ValidatingWebhookConfiguration/istiod.
ValidatingWebhookConfiguration/istiod-istio-system.
```

So why do we only need to watch one?

Reply:
Here  in the controller, it only reconciles one validatingwebhookconfiguration.

Reply:
The name is fixed

Reply:
I understand it is today. But I don't think it should be. If we install 2, we need to reconcile 2.

So we need to either install one, or reconcile both?

Reply:
I would argue each istiod should take charge of reconciling its own webhookconfiguration. Why does it need to reconcile another revision?

Reply:
OK, tried. I will revert this change. And I am sure we want to clean up in the future.